python/pyspark/context.py

```diff
@@ -565,9 +614,9 @@ def range(self, start, end=None, step=1, numSlices=None):
             end = start
             start = 0
 
-        return self.parallelize(range(start, end, step), numSlices)
+        return self.parallelize(list(range(start, end, step)), numSlices)
 
-    def parallelize(self, c, numSlices=None):
+    def parallelize(self, c: List[T], numSlices: Optional[int] = None) -> RDD[T]:
```

Comment:
We did you change the signature here? 

https://github.com/apache/spark/blob/ef4f2546c58ef5fe67be7047f9aa2a793519fd54/python/pyspark/context.pyi#L100